draft-rescorla-tls-esni.md

```diff
@@ -203,16 +203,26 @@ be used to encrypt the SNI for the associated domain name.
 The cipher suite list is orthogonal to the
 list of keys, so each key may be used with any cipher suite.
 
-This structure is placed in the RRData section of a TXT record as
-encoded above. The name of each TXT record MUST match the name composed
-of "_esni" and the query domain name. That is, if a client queries
-example.com, the ESNI TXT name is _esni.example.com.
+This structure is placed in the RRData section of a TXT record 
+as a base64-encoded string. If this encoding exceeds the 255 octet 
+limit of TXT strings, it must be split across multiple concatenated
+strings as per Section 3.1.3. of {{RFC4408}}.
+
```

Comment:
I read the text in RFC 4408 and I am a bit concerned that this is one of those things that does not work in theory but does work in practice. In theory, when there are multiple records of the same type for the same name, DNS servers or recursive resolvers can return the records in whichever order they see fit. If they actually do that, the concatenation becomes unpredictable.

Reply:
The text here suggests that we'd split up text within a single record, not across multiple records. Indeed, if the latter were the case, I don't think this would work.